C++/dfa.hpp

```diff
@@ -17,6 +17,7 @@ class DFA {
   map<string, vector<tuple<string, string, string>>> transitions;
   map<string, State*> states;
   int rejected;
+  int __num_start_states;
```

Comment:
__num_start_states should not be declared public, that's something that the class uses internally to check for the number of start states.